### DISCUSSION TITLE
Version test 0.6.1: Do we need to add keys manually for flax hddcoin and nchain?

### DISCUSSION THREAD

**Discussion by jimuyouyou** [guydavis/machinaris#350] [Q&A] [answered]
OS: ubuntu 20.0, docker-compose 1.25
machinaris version: test 0.6.1

I setup it according to the site https://www.machinaris.app/. Then I found flax, hddcoin and nchain are missing farmer process. After debugging into the docker /bin/bash, I noticed there is no keys for them. So I use flax/hddcoin/chia keys add method to manually import my key, and then restart the farmer. Everything works as expected now. However, I do have the doubt: is this(manually import keys) part of the design idea, or maybe there is something wrong with my setup?

BTW, chia works well by default, and it does not need the manual operation.

**Comment by guydavis** [comment, marked as answer]
Yes, apologies as this is not clear for Windows or Linux users who are just getting started, the steps are only mentioned explicitly in the Unraid app

In summary, the steps are:

1. From the [Machinaris install generator site](https://www.machinaris.app), save docker-compose.yml into a 'machinaris' folder.  
2. `docker-compose up -d`
3. Browse to http://localhost:8926 and complete key import or generate new.
4. Copy the just-created `~/.machinaris/mnemonic.txt` file to the other containers appdata folders such as `~/.machinaris-flax/`, `~/.machinaris-hddcoin`, etc
5. `docker-compose stop && docker-compose up -d` to have those containers restart and pickup the new key.

While steps # 1-3 are pretty clear, I will add messaging in the WebUI and wiki documentation to make clear that steps # 4-5 are needed.  Thanks for the suggestion.
